Quick_Deploy/JAX/model_repository/resnet50/1/model.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import triton_python_backend_utils as pb_utils
+import jax
+import jax.numpy as jnp
+import flaxmodels as fm
+
+import numpy as np
+from flax.jax_utils import replicate
+
+class TritonPythonModel:
+
+    def initialize(self, args):
+        self.key = jax.random.PRNGKey(0)
+        self.resnet18 = fm.ResNet18(output='logits', pretrained='imagenet')
+
+
+    def execute(self, requests):
+        responses = []
+        for request in requests:
+            inp = pb_utils.get_input_tensor_by_name(request, "image")
+            input_image = inp.as_numpy()
+
+            params = self.resnet18.init(self.key, input_image)
```

Comment:
I'm surprised by this `init` call here. I'm not overly familiar with how Flax or the Flaxmodels library structure things, but why do we call init here instead of in the `initialize` method. What does it actually do? Is it about needing to know the shape of the input image?

Reply:
Ah, good catch, I missed this.